Content.Shared/Silicons/StationAi/SharedStationAiSystem.Airlock.cs

```diff
@@ -1,25 +1,117 @@
 using Content.Shared.Doors.Components;
 using Robust.Shared.Serialization;
+using Content.Shared.Electrocution;
+using Content.Shared.Popups;
+using Content.Shared.Power.Components;
+using Robust.Shared.Audio;
 
 namespace Content.Shared.Silicons.StationAi;
 
 public abstract partial class SharedStationAiSystem
 {
+    private static readonly SoundPathSpecifier AirlockOverchargeDisabled = new("/Audio/Machines/airlock_overcharge_on.ogg");
+    private static readonly SoundPathSpecifier AirlockOverchargeEnabled = new("/Audio/Machines/airlock_overcharge_off.ogg");
+
     // Handles airlock radial
 
     private void InitializeAirlock()
     {
         SubscribeLocalEvent<DoorBoltComponent, StationAiBoltEvent>(OnAirlockBolt);
+        SubscribeLocalEvent<AirlockComponent, StationAiEmergencyAccessEvent>(OnAirlockEmergencyAccess);
+        SubscribeLocalEvent<ElectrifiedComponent, StationAiElectrifiedEvent>(OnElectrified);
     }
 
+    /// <summary>
+    /// Attempts to bolt door. If wire was cut (AI or for bolts) or its not powered - notifies AI and does nothing.
+    /// </summary>
     private void OnAirlockBolt(EntityUid ent, DoorBoltComponent component, StationAiBoltEvent args)
     {
+        if (!TryComp<StationAiWhitelistComponent>(ent, out var whiteList))
+        {
+            return;
+        }
+
+        if (!whiteList.Enabled || component.BoltWireCut || !component.Powered)
+        {
+            _popup.PopupClient(Loc.GetString("ai-device-not-responding"), args.User, PopupType.MediumCaution);
+            return;
+        }
+
         _doors.SetBoltsDown((ent, component), args.Bolted, args.User, predicted: true);
     }
+
+    /// <summary>
+    /// Attempts to bolt door. If wire was cut (AI) or its not powered - notifies AI and does nothing.
+    /// </summary>
+    private void OnAirlockEmergencyAccess(EntityUid ent, AirlockComponent component, StationAiEmergencyAccessEvent args)
+    {
+        if (!TryComp<StationAiWhitelistComponent>(ent, out var whiteList))
+        {
+            return;
+        }
+
+        if (!whiteList.Enabled || !component.Powered)
+        {
+            _popup.PopupClient(Loc.GetString("ai-device-not-responding"), args.User, PopupType.MediumCaution);
+            return;
+        }
+
```

Comment:
AirlockComponent.Powered is obsolete please don't use it.

Reply:
used PowerReceiver.IsPowered instead. we might need to mark it obsolete and remove related code btw... its really misleading...

Reply:
is it true for bolts component?

Reply:
fixed, and used PowerReceiver system